plugins/modules/vmware_vmotion.py

```diff
@@ -168,67 +198,92 @@ def __init__(self, module):
+        if dest_cluster_name is not None:
+            self.cluster_object = find_cluster_by_name(content=self.content,
+                                                       cluster_name=dest_cluster_name, datacenter=datacenter_object)
+            if self.cluster_object:
+                self.cluster_hosts = []
+                for host in self.cluster_object.host:
+                    self.cluster_hosts.append(host)
+            else:
+                self.module.fail_json(msg="Unable to find destination cluster %s" % dest_cluster_name)
 
-        # Get Datacenter if specified by user
-        dest_datacenter = self.destination_datacenter
-        if dest_datacenter is not None:
-            datacenter_object = find_datacenter_by_name(content=self.content, datacenter_name=dest_datacenter)
-            if datacenter_object:
-                dest_datacenter = datacenter_object
-
-        # Get Destination Datastore if specified by user
+        # Get Destination Datastore or Datastore Cluster if specified by user
         dest_datastore = self.params.get('destination_datastore', None)
+        dest_datastore_cluster = self.params.get('destination_datastore_cluster', None)
+
+        if dest_datastore and dest_datastore_cluster:
+            self.module.fail_json(msg="Please only define one: destination_datastore or destination_datastore_cluster")
+
         self.datastore_object = None
+        self.datastore_cluster_object = None
+
         if dest_datastore is not None:
             self.datastore_object = find_datastore_by_name(content=self.content,
-                                                           datastore_name=dest_datastore, datacenter_name=dest_datacenter)
-
-        # At-least one of datastore, host system is required to migrate
-        if self.datastore_object is None and self.host_object is None:
-            self.module.fail_json(msg="Unable to find destination datastore"
-                                      " and destination host system.")
-
-        # Get Destination resourcepool
-        dest_resourcepool = self.params.get('destination_resourcepool', None)
-        self.resourcepool_object = None
-        if dest_resourcepool:
-            self.resourcepool_object = find_resource_pool_by_name(content=self.content,
-                                                                  resource_pool_name=dest_resourcepool)
-            # Fail if resourcePool object is not found
-            if self.resourcepool_object is None:
-                self.module.fail_json(msg="Unable to find destination resource pool object for %s." % dest_resourcepool)
-        elif not dest_resourcepool and dest_host_name:
-            self.resourcepool_object = self.host_object.parent.resourcePool
+                                                           datastore_name=dest_datastore,
+                                                           datacenter_name=dest_datacenter)
+                                           
```

Comment:
```suggestion

```

Looks like something in our CI doesn't like the spaces here:

plugins/modules/vmware_vmotion.py:249:1: W293: blank line contains whitespace
plugins/modules/vmware_vmotion.py:249:0: trailing-whitespace: Trailing whitespace